codegen/type_resolution.py

```diff
@@ -147,17 +142,21 @@ def pattern_match_impl(
 
 @dataclass(frozen=True)
 class GenericValueReference(CompileTimeConstant):
-    argument_name: str
+    name: str
+
+    @property
+    def argument(self) -> GenericArgument:
+        return GenericArgument(self.name, True)
 
     def format_for_output_to_user(self) -> str:
-        return f"@{self.argument_name}"
+        return self.name
```

Comment:
Is the `@` embedded in the name now?

Reply:
I think its been embedded for as long as we've been parsing it like that (always?). This is just a user-facing bug that became obvious now we're using the `UnresolvedType`s for error messages.

Reply:
ok, I see